server/cromwell.py

```diff
@@ -0,0 +1,247 @@
+            service_account_email = service_account_dict['client_email']
+            intermediate_dir = f'gs://cpg-{dataset}-{access_level}-tmp/cromwell'
+
+            if not service_account_json or not intermediate_dir:
+                raise web.HTTPBadRequest(
+                    reason=f'Invalid access level "{access_level}"'
+                )
+
+            hail_bucket = f'cpg-{dataset}-hail'
+            backend = hb.ServiceBackend(
+                billing_project=dataset,
+                bucket=hail_bucket,
+                token=hail_token,
+            )
+
+            commit = params['commit']
+            if not commit or commit == 'HEAD':
+                raise web.HTTPBadRequest(reason='Invalid commit parameter')
+
+            libs = params.get('dependencies')
+            if not isinstance(libs, list):
+                raise web.HTTPBadRequest(reason='Expected "dependencies" to be a list')
+            cwd = params['cwd']
+            wf = params['workflow']
+            if not wf:
+                raise web.HTTPBadRequest(reason='Invalid script parameter')
+
+            input_jsons = params.get('input_json_paths', [])
+            input_dict = params.get('inputs_dict')
+
+            # This metadata dictionary gets stored at the output_dir location.
+            timestamp = datetime.now().astimezone().isoformat()
+            metadata = json.dumps(
+                get_analysis_runner_metadata(
+                    timestamp=timestamp,
+                    dataset=dataset,
+                    user=email,
+                    access_level=access_level,
+                    repo=repo,
+                    commit=commit,
+                    script=wf,
+                    description=params['description'],
+                    output=output_dir,
+                    driver_image=DRIVER_IMAGE,
+                    cwd=cwd,
+                    mode='cromwell',
+                )
+            )
+
+            # Publish the metadata to Pub/Sub. Wait for the result before running the batch.
+            publisher.publish(PUBSUB_TOPIC, metadata.encode('utf-8')).result()
+
+            user_name = email.split('@')[0]
+            batch_name = f'{user_name} {repo}:{commit}/cromwell/{wf}'
+
+            batch = hb.Batch(backend=backend, name=batch_name)
+
+            job = batch.new_job(name='driver')
+            job = prepare_git_job(
+                job=job,
+                dataset=dataset,
+                access_level=access_level,
+                repo=repo,
+                commit=commit,
+                metadata_str=metadata,
+                output_dir=output_dir,
+                print_all_statements=False,
+            )
+            job.env('OUTPUT', output_dir)
+            # we'll need to put the commands like this, because
+            # the job.command is individually scoped, ie, the cd
+            # doesn't carry across the different job.commands
```

Comment:
I'm pretty sure that's not the case: https://github.com/populationgenomics/analysis-runner/blob/main/server/main.py#L160

Reply:
Yep you're right, the original change that prompted this was actually for an unrelated reason, I'll revert this bit.